doc/user_interface.md

```diff
@@ -0,0 +1,616 @@
+# User interface
+
+## General principles
+
+The Visual Query Builder interface breaks down into 2 panels:
+
+- the **transformation pipeline on the left**: this is where you can see the
+  ordered series of transformation steps that are applied to your data. When you
+  need to configure a given step, the left panel switches to the **step edition
+  form**
+
+- the **data table on the right**: this is where you can click on you data to apply
+  transformations. The right panel also includes **widgets** above the table.
+  Transformations are available from the columns headers, from the widgets or
+  can be found via the **search bar**.
+
+<img src="./img/general_interface.jpg" width="800" />
+</br>
+</br>
+<img src="./img/column_header_menu.jpg" height="300" />
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
+<img src="./img/widget_menu.jpg" height="300" />
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
+<img src="./img/search_bar_menu.jpg" height="300" />
+</br>
+</br>
+<img src="./img/step_edition_mode.jpg" width="800" />
+
+Everytime you interact with the table or widgets to transform data, the
+corresponding step is logged in the transformation pipeline. Any step can be
+edited or deleted.
+If a step in the middle of the pipeline gets edited or deleted, the following
+steps remain unchanged.
+
+If you click on any step of the pipeline, the data table on the right will
+update to show you the result of executing the the transformations step until
```

Comment:
Just a suggestion to ligthen up the wording:
```suggestion
update to show you the result of the transformations step until
```

Reply:
Ok thx